src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/InitializeHandler.cs

```diff
@@ -92,7 +92,7 @@ internal class InitializeHandler : IRequestHandler<InitializeParams, InitializeR
                     HoverProvider = true,
                     DefinitionProvider = true,
                     DocumentHighlightProvider = true,
-                    RenameProvider = true,
+                    RenameProvider = !_languageServerFeatureOptions.SingleServerRenameSupport,
```

Comment:
For readability I'd break this into its own if statement so more features can make use of it (kind of referencing my previous comment of having a general "single server support flag")